### PR TITLE
Get rewards by UserID and e2e tests

### PR DESCRIPTION
**Pull request by 0xdev22**

### CODE REVIEW

File: gqlgen.yml [hunk 1]
```diff
@@ -143,6 +143,13 @@ models:
     extraFields:
       AftermarketDeviceID:
         type: "int"
+  UserRewards:
+    fields:
+      history:
+        resolver: true
+    extraFields:
+      User:
+        type: "*github.com/ethereum/go-ethereum/common.Address"
```
Comment (elffjs) [inline gqlgen.yml:152]: Must it be a pointer?

File: internal/repositories/rewards.go [hunk 2]
```diff
@@ -106,9 +107,15 @@ func (r *Repository) paginateRewards(ctx context.Context, conditions []qm.QueryM
 	for i, rw := range all {
 		reward := RewardToAPI(*rw)
 
+		crsr, err := rwCursorHelper.EncodeCursor(RewardsCursor{
+			Week: reward.Week,
```
Comment (elffjs) [inline internal/repositories/rewards.go:111]: I think we'd better actually fill in each field of the cursor.

File: internal/repositories/rewards.go [hunk 3]
```diff
@@ -244,3 +251,129 @@ func (r *Repository) PaginateAftermarketDeviceEarningsByID(ctx context.Context,
...
+	}
+
+	return &gmodel.UserRewards{
+		TotalTokens: dbtypes.NullDecimalToInt(summary.TokenSum),
+		History:     earningsConn,
+		User:        &user,
+	}, nil
+}
+
+var rewardsColumnsTuple = "(" + models.RewardColumns.IssuanceWeek + ", " + models.RewardColumns.VehicleID + ")"
```
Comment (elffjs) [inline internal/repositories/rewards.go:279]: Can we be more specific? `rewardsCursorColumns`?

File: internal/repositories/rewards.go (hunk 2 above)
Comment (elffjs) [reply internal/repositories/rewards.go:111]: I'm actually confused by us needing to encode this twice.

File: internal/repositories/rewards.go [hunk 4]
```diff
@@ -244,3 +251,129 @@ func (r *Repository) PaginateAftermarketDeviceEarningsByID(ctx context.Context,
...
+
+func (r *Repository) PaginateGetEarningsByUsersDevices(ctx context.Context, userDeviceEarnings *gmodel.UserRewards, first *int, after *string, last *int, before *string) (*gmodel.EarningsConnection, error) {
```
Comment (elffjs) [inline internal/repositories/rewards.go:281]: I guess my question is—why not go all the way and use `paginateRewards`?

File: internal/repositories/rewards.go [hunk 5]
```diff
@@ -244,3 +255,48 @@ func (r *Repository) PaginateAftermarketDeviceEarningsByID(ctx context.Context,
...
+	}
+
+	afd, err := r.paginateRewards(ctx, queryMods, first, after, last, before, limit)
+	if err != nil {
+		return nil, err
+	}
+
+	userDeviceEarnings.History.Edges = afd.Edges
+	userDeviceEarnings.History.Nodes = afd.Nodes
+	userDeviceEarnings.History.PageInfo = afd.PageInfo
```
Comment (elffjs) [inline internal/repositories/rewards.go:300]: This is still a weird construction. I hope we will get rid of it someday.

**Review by elffjs — APPROVED** [review]
Very complete.

**Comment by elffjs** [comment]
@Allyson-English Could you take a look? You have some experience with this with DCN.

**Review by Allyson-English — APPROVED** [review]

**Pull request merged**